### ISSUE TITLE
[Request] GameShark CDX Support

### ISSUE THREAD

**Issue by 60fpshacksrock** [xjsxjs197/WiiSXRX_2022#43]
Can support for pcsx-r's "CDX_Dongle.bin" be added? 
This allows the GameShark 3.3 (and maybe 3.4) CDX to be used on the emulator.

**Comment by eku** [comment]
Have you tried GameShark Version 5.0 ISO ROM instead?

**Comment by 60fpshacksrock** [comment]
Yes. It is a good one, but GS V5 has encrypted codes whereas the older GS version use unencrypted codes and can be directly edited. Not to mention the GS 3.3 CDX uses, if pcsxr is correct, a 256KB custom memory card image so there is much more room for codes.

The GS 3.3 CDX has a nicer UI as well. Its UI resembles the older GS cart that could be used on the older PSX's with the parallel port.

**Comment by xjsxjs197** [comment]
It is verified that GameShark 3.3 support can be added. 
However, the dynamic compilation instruction BEQ needs special modification.
The cdrom label of my GS 3.3 is 990716_ 1514, I don't know whether this ID can be used to identify GS 3.3.
The cdromId is ARCDEXE, which is definitely not possible, because other versions of GS are the same

**Comment by 60fpshacksrock** [comment]
Maybe checksum ARCDEXE?

**Comment by saulfabregwiivc** [comment]
reliving old issue

@60fpshacksrock 

Could you test this build i've compiled until git https://github.com/xjsxjs197/WiiSXRX_2022/commit/cc9d199 ?
[WiiStation_GameSharkCDXMerge_git_cc9d199_13112022.zip](https://github.com/xjsxjs197/WiiSXRX_2022/files/10828444/WiiStation_GameSharkCDXMerge_git_cc9d199_13112022.zip)

**Comment by 60fpshacksrock** [comment]
@saulfabregwiivc 

My apologies for not replying back to you. I had forgotten that I even posted this. Yes, with this build it does now boot and get "in-game" to the cheat menus so to speak. 

But disc swapping doesn't seem to work as it seems to freeze, and requires you to press reset on the Wii to get the menu open as X+Y is irresponsive.
Can you merge this with the latest 2024 builds?

**Comment by 60fpshacksrock** [comment]
> It is verified that GameShark 3.3 support can be added. However, the dynamic compilation instruction BEQ needs special modification. The cdrom label of my GS 3.3 is 990716_ 1514, I don't know whether this ID can be used to identify GS 3.3. The cdromId is ARCDEXE, which is definitely not possible, because other versions of GS are the same

Is it possible to add the changes made by @saulfabregwiivc in that posted woriking .dol to the newest builds?
